src/lib/fragments/Molecule3dViewer.js

```diff
@@ -14,9 +14,16 @@ export default class Molecule3dViewer extends Component {
     }
 
     onRenderNewData(glviewer) {
-        glviewer.center();
-        const zoomRatio = 0.8;
-        glviewer.zoom(zoomRatio);
+        glviewer.zoomTo(
+            this.props.zoomTo.sel,
+            this.props.zoomTo.animationDuration,
+            this.props.zoomTo.fixedPath
+        );
+        glviewer.zoom(
+            this.props.zoom.factor,
+            this.props.zoom.animationDuration,
+            this.props.zoom.fixedPath
+        );
```

Comment:
This looks great for initial render, but I'm trying to figure out what will happen on various kinds of updates.

According to https://www.npmjs.com/package/molecule-3d-for-react:

> onRenderNewData {function} [(glviewer) => {}]
> A callback for when the modelData has changed and it has been re-rendered in the viewer.

So we're only going to get here when `modelData` changes, even though many other props may have changed too? Or do we actually get here on any rerender?

Specifically here are two cases I'm interested in:
1. The app has a toggle to turn some features on/off - labels or orbitals perhaps - and the user rotates & zooms (also I just discovered you can pan with ctrl-drag) then toggles those features. Do we snap back to the original view? We shouldn't, we should keep the view they found.
2. The app has a "reset zoom" button, or a button to zoom to a particular view, that just gives new `zoom` and/or `zoomTo` props but leaves all the other props unchanged. Does this work? ie the user zooms, rotates, and pans, does this button get back to the desired state?

Reply:
I think you've raised a good point here, and this might be require changes to the behaviour of the component when it updates. So far, I think the way we've handled updates to props with Molecule3dViewer is by re-rendering the container and updating the parent either `div` or `dcc.Loading` with our callbacks. 

I'm noticing that if we do attempt to update props directly with a callback, the component doesn't re-render as it should, and the loading state is stuck.

The following snippet shows this when the button is clicked to change the `backgroundColor` prop - the prop changes, but the loading state remains `true`. 

```
import json
import six.moves.urllib.request as urlreq
from six import PY3

import dash
import dash_bio as dashbio
import dash_html_components as html
import dash_core_components as dcc

app = dash.Dash(__name__)

model_data = urlreq.urlopen(
    'https://raw.githubusercontent.com/plotly/dash-bio-docs-files/master/' +
    'mol3d/model_data.js'
).read()
styles_data = urlreq.urlopen(
    'https://raw.githubusercontent.com/plotly/dash-bio-docs-files/master/' +
    'mol3d/styles_data.js'
).read()

if PY3:
    model_data = model_data.decode('utf-8')
    styles_data = styles_data.decode('utf-8')

model_data = json.loads(model_data)
styles_data = json.loads(styles_data)

app.layout = html.Div([
    dashbio.Molecule3dViewer(
        id='my-dashbio-molecule3d',
        styles=styles_data,
        modelData=model_data
    ),
    html.Button(id="button", n_clicks=0, children="Toggle background")
])


@app.callback(
    dash.dependencies.Output('my-dashbio-molecule3d', 'backgroundColor'),
    [dash.dependencies.Input('button', 'n_clicks')],
    prevent_initial_call=True
)
def toggle_labels(n_clicks):
    if n_clicks > 0:
        return("#FF0000")


if __name__ == '__main__':
    app.run_server(debug=True)
```

It doesn't look like our `render()` method would need any other data, so I'm a little confused why it wouldn't re-render by default.

Reply:
Sorry, false alarm, this doesn't seem to be the case, it responds correctly to re-renders, the component was just missing a condition in `shouldComponentUpdate` for `labels`. For the two app scenarios, here's what we have:

1. If we rotate/zoom, and then update or toggle the props (eg.`backgroundColor` or `labels`), the present view remains the same

2. This one doesn't work. I think, as you pointed out, the `glviewer` view will only change on initial render. I'll try to find a workaround for this, but it might involve adding a method to the Molecule3DViewer component itself.

Reply:
You should be able to do it without modifying `Molecule3d`
- store `glviewer` in the instance so you have it available (`this.glviewer = glviewer` during `onRenderNewData`)
- store the `zoom` and `zoomTo` props similarly after every `glviewer.zoomTo/zoom` chain.
- add a `componentDidUpdate` method that checks if `zoom` and `zoomTo` are `===` their previous values; if they've changed (and you have a `glviewer`), `this.glviewer.zoomTo/zoom` and store the props again.

Reply:
Thanks for the suggestion, added in aa7bbfe.

One question - Is there any particular reason we need to store the `zoom` or `zoomTo` values? We can access them with `prevProps` in `componentDidUpdate` for the same functionality so I'm confused why we'd need to store them again.

Reply:
Oh I forgot about that - good call!